### PR TITLE
Rx Specific lua params

### PR DESCRIPTION
**Pull request by StonedDawg**
add ability to change some RX specific parameters, such as Antenna diversity capable RX doesn't have to rebuild firmware to activate or deactivate Diversity mode.

edit : 
this PR now adds these parameter to each RX : 
1. antenna Mode (for diversity RX only) : antenna B, antenna A, diversity
2. enable selected RX wifi (while the one in ``wifi connectivity > enable RX wifi``, is a broadcast command to all rx that are powered on)
3. telemetry power to rx that has a PA
4. firmware version and hash

### CODE REVIEW

**Comment by pkendall64** [comment]
Do you think we can change the POWER_MIN/MAX to an array like the TX module. I know that some RX's have 4 values that can be "configured". And have a default power setting as well.

**Comment by StonedDawg** [comment]
@pkendall64 well, our initial discussion was just about "low power" and "standard" for RXes option, but yeah now I see some RXes has 4 calibrated value already... i'll get on that once I figured out why my config.commit crashes it :(

**Comment by CapnBry** [comment]
I'm all for Lua params for the RX, but you 100% need to not copy paste code from the TX to make new code out of it. All the Lua functions you've copied we'll now have in two places. Every line of code that's identical or serves the same function needs to be common to both code bases, not copy pasted and possibly modified.

I also have a ton of other config changes pending a config version change so I'd like the config storage to be separate so we can get them all at once. We can make the config PR first then you can apply the Lua on top of that once the storage is in place.
https://github.com/orgs/ExpressLRS/projects/2

File: src/lib/POWERMGNT/POWERMGNT.h [hunk 1]
```diff
@@ -15,16 +15,17 @@
 #include <nvs.h>
 #endif
 
-#if defined(TARGET_RX)
+#if defined(HighPower) && !defined(UNLOCK_HIGHER_POWER)
```
Comment (pkendall64) [inline src/lib/POWERMGNT/POWERMGNT.h:18]: I think this was wrong before and still is!
Shouldn't this be
```cpp
#if defined(HighPower) && defined(UNLOCK_HIGHER_POWER)
```

Comment (StonedDawg) [reply src/lib/POWERMGNT/POWERMGNT.h:18]: it is confusing but this seems correct. HighPower is not the max Power, so if it is not unlocked, then maxPower will only get as high as HighPower?

Comment (pkendall64) [reply src/lib/POWERMGNT/POWERMGNT.h:18]: > it is confusing but this seems correct. HighPower is not the max Power, so if it is not unlocked, then maxPower will only get as high as HighPower?

Yep very confusing, `UNLOCK_HIGHER_POWER` unlocks `MaxPower` if `HighPower` is defined.

**Comment by StonedDawg** [comment]
@CapnBry I think this is ready, i'm good with this now or wait for your upcoming config changes

File: src/src/rx_main.cpp [hunk 2]
```diff
@@ -1225,6 +1241,31 @@ void loop()
...
+        Radio.Resume();
+        /*
+        // Currently this code is not aligining the FHSSptr correctly so it times out and get LostConnection
+        while((micros() - start) % ExpressLRS_currAirRate_Modparams->interval > 50) ;
+        uint32_t offset = micros() - start;
+        int intervals = offset / ExpressLRS_currAirRate_Modparams->interval - 1;
+        NonceRX = (NonceRX + intervals) % ExpressLRS_currAirRate_Modparams->FHSShopInterval;
+        FHSSptr = (FHSSptr + (intervals / ExpressLRS_currAirRate_Modparams->FHSShopInterval)) % FHSS_SEQUENCE_CNT;
+        Radio.SetFrequencyReg(FHSSgetNextFreq());
+        */
```
Comment (pkendall64) [inline src/src/rx_main.cpp:1259]: @CapnBry can you figure out why this doesn't work? Does my math just suck that hard!

**Review by pkendall64 — APPROVED** [review]

**Comment by pkendall64** [comment]
I added a commit that make the commit section a lot cleaner and doesn't require changes to the radio driver code.
So I reverted those parts and fixed it.

**Comment by StonedDawg** [comment]
cool! thanks @pkendall64 !

**Review by JyeSmith — COMMENTED** [review]
Tested with EP and Zorro.

When starting rx wifi lua will lock up.  This is probably because the rx is no longer responsive.  Can we force a lua refresh back to the device menu?

**Comment by pkendall64** [comment]
> Tested with EP and Zorro.
> 
> When starting rx wifi lua will lock up. This is probably because the rx is no longer responsive. Can we force a lua refresh back to the device menu?

I ran into this with the "Loan Model" code I'm working on. I think whats happening is the command is sent to the RX, but the LUA then goes into polling mode and the packets can't be delivered to the MSP sender is stuck. Probably the best solution is for the TX to reset the MSP sender on disconnect.

**Review by JyeSmith — APPROVED** [review]

**Comment by RM-FPV** [comment]
Hi!
Is that already working?

**telemetry power to rx**

Can't see any rx settings in current FW

**Comment by StonedDawg** [comment]
@RM-FPV it's in master branch already.

**Comment by RM-FPV** [comment]
Is it in V2.4.0 ?
I couldn't see it was mentioned in any release.

**Comment by StonedDawg** [comment]
@RM-FPV if you don't know how to flash master branch, it would be better to stick with the released version (or jump to discord and we can help you if you really want to use master). anyway this will be available in 3.0.

**Pull request merged**